ols/src/cache/redis_cache.py

```diff
@@ -35,7 +35,7 @@ def initialize_redis(self, config: RedisConfig) -> None:
         if config.password is not None:
             kwargs["password"] = config.password
         if config.ca_cert_path is not None:
-            kwargs["ssl"] = True
+            kwargs["ssl"] = True  # type: ignore
```

Comment:
what's the issue here?  should we be setting types on kwargs instead?  (type Any?)

cc @tisnik

Reply:
I am running into this error in one of the verify steps
```
mypy --explicit-package-bases --disallow-untyped-calls --disallow-untyped-defs --disallow-incomplete-defs ols/
ols/src/cache/redis_cache.py:38: error: Incompatible types in assignment (expression has type "bool", target has type "str")  [assignment]
```
So I cross checked the redis dependency version and source code: https://redis.readthedocs.io/en/latest/_modules/redis/client.html, it says `bool` but I am not sure why our CI verify step is expecting it to be `str`. Same thing doesn't happen in my local again. I tried aligning the dependency versions with mainstream, failed in them and finally ended up putting an ignore comment beside it.

Reply:
I assume it's because you're also putting string content into the dict, so mypy expects the all the types to be the same.  you probably need to give type hints when you construct the dict to indicate the type is str OR bool.

again @tisnik can probably help better than me here

Reply:
i think it would be something like `a: dict[str, str|bool] = {}`

Reply:
yes exactly. Mypy is doing type inference (like in ML, OCaml, F# and in less ideal way in Go). So in this case you need to specify explicitly that kwargs is `dict[str, Any]` or `dict[str, str | bool]`

Reply:
ah @bparees was faster ;)

Reply:
Updated. Thanks